src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,77 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import React, { useEffect, useState } from 'react';
+
+type Props = {
+  filterTodos: (query: string, groupBy: string) => void;
+};
+
+export const TodoFilter: React.FC<Props> = ({ filterTodos }) => {
+  const [query, setQuery] = useState('');
+  const [groupBy, setGroupBy] = useState('all');
+
+  useEffect(() => {
+    filterTodos(groupBy, query);
+  }, [query, groupBy]);
+
+  const handleChange = (event: React.ChangeEvent<HTMLSelectElement>
+  | React.ChangeEvent<HTMLInputElement>) => {
+    const { value, localName } = event.target;
+
+    if (localName === 'input') {
+      setQuery(value);
+    }
+
+    if (localName === 'select') {
+      setGroupBy(value);
+    }
+  };
+
+  const resetQuery = () => {
+    setQuery('');
+  };
+
+  return (
+    <form
+      className="field has-addons"
+    >
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            onChange={handleChange}
+            value={groupBy}
+          >
+            <option value="all">All</option>
+            <option value="active">Active</option>
+            <option value="completed">Completed</option>
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={query}
+          onChange={handleChange}
         />
-      </span>
-    </p>
-  </form>
-);
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        {query.trim().length > 0 ? (
+          <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
```

Comment:
to get rid of this comment you could use aria-label with some descriptive text for the button